pkg/proxy/proxy_test.go

```
@@ -0,0 +1,96 @@
+	}
+	defer clientConn.Close()
+	defer clientEnd.Close()
+	defer server.Close()
+	// Write on the mock server
+	go func() {
+		serverConn, err := server.Accept()
+		if err != nil {
+			t.Errorf("Failed to accept connection: %v", err)
+			return
+		}
+		serverConn.Write([]byte("HTTP/1.1 200 OK\r\nContent-Length: 11\r\n\r\nHello world"))
```

golangci-lint (1.62.2)

Error return value of `serverConn.Write` is not checked

(errcheck)